### PR TITLE
[O2B-1271] Store QC flags' origin from gRPC messages

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- NA

Notable changes for developers:
- NA

Changes made to the database:
- Added `origin` column to `quality_control_flags` table

### CODE REVIEW

File: lib/public/views/QcFlags/ActiveColumns/qcFlagsActiveColumns.js [hunk 1]
```diff
@@ -57,7 +57,9 @@ export const qcFlagsActiveColumns = {
     createdBy: {
         name: 'Created by',
         visible: true,
-        format: (createdBy) => createdBy?.name || '-',
+        format: (createdBy, { origin }) => createdBy?.name === 'Anonymous' || !createdBy?.name
+            ? origin || createdBy?.name || '-'
+            : createdBy.name,
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/QcFlags/ActiveColumns/qcFlagsActiveColumns.js]: applied

File: lib/public/views/QcFlags/ActiveColumns/qcFlagsActiveColumns.js [hunk 2]
```diff
@@ -14,6 +14,7 @@
 import { h } from '/js/src/index.js';
 import { formatTimestamp } from '../../../utilities/formatting/formatTimestamp.js';
 import { qcFlagTypeColoredBadge } from '../common/qcFlagTypeColoredBadge.js';
+import { formatQcFlagCreatedBy } from '../common/formatQcFlagCreatedBy.js';
```
Comment (martinboulais) [inline lib/public/views/QcFlags/ActiveColumns/qcFlagsActiveColumns.js:17]: This should probably not go into common directory. Same for qcFlagTypeColoredBadge, I have not noticed before

File: lib/public/views/QcFlags/common/formatQcFlagCreatedBy.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+/**
+ * Format QC flag creator name
+ * @param {QcFlag} qcFlag QC flag
```
Comment (martinboulais) [inline lib/public/views/QcFlags/common/formatQcFlagCreatedBy.js:17]: QC flag of which creator name should be formatted

Please put a new line before the first @param

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/public/views/QcFlags/common/formatQcFlagCreatedBy.js (hunk 3 above)
Comment (xsalonx) [reply lib/public/views/QcFlags/common/formatQcFlagCreatedBy.js:17]: applied

**Review by martinboulais — APPROVED** [review]

**Pull request merged**